tests/query_string_tests.rb

```diff
@@ -1,57 +1,67 @@
 Shindo.tests('Excon query string variants') do
   with_rackup('query_string.ru') do
-    connection = nil
+    connection = Excon.new('http://127.0.0.1:9292')
 
     tests(":query => {:foo => 'bar'}") do
-      tests("query string sent").returns('foo=bar') do
-        connection = Excon.new('http://127.0.0.1:9292')
-
-        response = connection.request(:method => :get, :path => '/query', :query => {:foo => 'bar'})
-        query_string = response.body[7..-1] # query string sent
+      response = connection.request(:method => :get, :path => '/query', :query => {:foo => 'bar'})
+      query_string = response.body[7..-1] # query string sent
 
+      tests("query string sent").returns('foo=bar') do
         query_string
       end
     end
 
     tests(":query => {:foo => nil}") do
-      tests("query string sent").returns('foo') do
-        response = connection.request(:method => :get, :path => '/query', :query => {:foo => nil})
-        query_string = response.body[7..-1] # query string sent
+      response = connection.request(:method => :get, :path => '/query', :query => {:foo => nil})
+      query_string = response.body[7..-1] # query string sent
 
+      tests("query string sent").returns('foo') do
         query_string
       end
     end
 
     tests(":query => {:foo => 'bar', :me => nil}") do
-      query_string = nil
+      response = connection.request(:method => :get, :path => '/query', :query => {:foo => 'bar', :me => nil})
+      query_string = response.body[7..-1] # query string sent
 
       test("query string sent includes 'foo=bar'") do
-        response = connection.request(:method => :get, :path => '/query', :query => {:foo => 'bar', :me => nil})
-        query_string = response.body[7..-1] # query string sent
-
         query_string.split('&').include?('foo=bar')
       end
 
       test("query string sent includes 'me'") do
         query_string.split('&').include?('me')
       end
-
     end
 
     tests(":query => {:foo => 'bar', :me => 'too'}") do
-      query_string = nil
+      response = connection.request(:method => :get, :path => '/query', :query => {:foo => 'bar', :me => 'too'})
+      query_string = response.body[7..-1] # query string sent
 
       test("query string sent includes 'foo=bar'") do
-        response = connection.request(:method => :get, :path => '/query', :query => {:foo => 'bar', :me => 'too'})
-        query_string = response.body[7..-1] # query string sent
-
         query_string.split('&').include?('foo=bar')
       end
 
       test("query string sent includes 'me=too'") do
         query_string.split('&').include?('me=too')
       end
+    end
+
+    # To emulate the Rails and PHP style of serializing an array, use { 'foo[]' => [ array elements ] }
+    tests(":query => {:foo => ['bar', 'baz'], :me => 'too'}") do
```

Comment:
Maybe, for full clarity we could just have the test use the `[]` style explicitly? That way it would be clearer the purpose for this test. What do you think?